src/twisted/conch/ssh/keys.py

```diff
@@ -26,6 +27,8 @@
     load_pem_private_key,
     load_ssh_public_key,
 )
+from nacl.exceptions import BadSignatureError  # type: ignore[import]
```

Comment:
I think that for a soft-dependency this needs to be wrapped in a `try/except ImportError` and define `BadSignatureError = object` or something like this

I am no what  is best way to implement it as a soft-dependency.
Because if you have it as a soft-dependency  and pynal is not installed and you have an older OpenSSL  you will not have support for ED25519

Reply:
Oh, right, good point.  I'll sort this out as soon as I get a moment.

Reply:
No hurry from my side. I did a quick review and the code looks very well written.
I am looking to do a 20.7.0 release ASAP...but we can also do a 20.8.0 release to include this PR.